pkg/config/analysis/analyzers/gateway/gateway.go

```diff
@@ -50,7 +50,7 @@ func (*IngressGatewayPortAnalyzer) Metadata() analysis.Metadata {
 
 // Analyze implements analysis.Analyzer
 func (s *IngressGatewayPortAnalyzer) Analyze(c analysis.Context) {
-	c.ForEach(collections.IstioNetworkingV1Alpha3Gateways.Name(), func(r *resource.Instance) bool {
+	c.ForEachNeedsAnalyze(collections.IstioNetworkingV1Alpha3Gateways.Name(), func(r *resource.Instance) bool {
```

Comment:
This (and other places) doesn't really seem safe. The analyzer actually depends on Gateway+Pod+Services interaction, but we only run it on Gateway change. Really if the Service changes, we should update it as well

Reply:
Revised, now if service A has changed, the entire lists of resources from Gateway, Pod and Service collections will be analyzed to entire the statuses are always synced.